fattmerchant-ios-sdk/Cardpresent/Usecase/TakePayment.swift

```diff
@@ -94,16 +94,17 @@ class TakePayment {
     return (job, nil)
   }
 
-  /// Creates a ChargeRequest from an Amount and a PaymentMethod id
+  /// Creates a ChargeRequest from an Amount and a PaymentMethod id, poNumber is optional
   /// - Parameters:
   ///   - amount: the Amount to charge
   ///   - paymentMethodId: the id of the PaymentMethod to charge
+    /// - poNumber: The purchase order number for the transaction
   /// - Returns: a ChargeRequest
-  internal static func createChargeRequest(amount: Amount, paymentMethodId: String) -> ChargeRequest {
+    internal static func createChargeRequest(amount: Amount, paymentMethodId: String, poNumber: String?  ) -> ChargeRequest {
     let chargeRequestMeta = [
-      "subtotal": amount.dollarsString()
+      "subtotal": amount.dollarsString(),
+      "poNumber": poNumber ?? ""
```

Comment:
This isn't the correct use case, if there is no `poNumber` passed in, don't set anything, not even an empty string. It should be something like this

```
let chargeRequestMeta = ["subtotal": amount.dollarsString()]

if poNumber != null
  chargeRequestMeta.poNumber = poNumber
```